model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -147,6 +148,37 @@ class ModelReplicationServer(
             call.respondDelta(versionHash, baseVersionHash)
         }
 
+        delete<Paths.deleteRepositoryBranch> {
+            val repositoryName = call.parameters["repository"]
+            if (repositoryName == null) {
+                call.respondText("Request lacks repository name", status = HttpStatusCode.BadRequest)
+                return@delete
+            }
+            val repositoryId = try {
+                RepositoryId(repositoryName)
+            } catch (e: IllegalArgumentException) {
+                call.respondText("Invalid repository name '$repositoryName'", status = HttpStatusCode.BadRequest)
+                return@delete
+            }
+            val branch = call.parameters["branch"]
+            if (branch == null) {
+                call.respondText("Request lacks branch name", status = HttpStatusCode.BadRequest)
+                return@delete
+            }
+
+            if (!repositoriesManager.getBranchNames(repositoryId).contains(branch)) {
+                call.respondText(
+                    "Repository does not exist or branch '$branch' does not exist in repository '$repositoryId'",
+                    status = HttpStatusCode.NotFound,
+                )
+                return@delete
+            }
+
+            repositoriesManager.removeBranches(repositoryId, setOf(branch))
```

Comment:
I suspect this only deletes a branch in the V2 storage and not the V1 storage. I will take look at this and come back to you.

Reply:
Never mind. This is already correctly handled.